src/Builder.php

```diff
@@ -11,277 +11,70 @@
 
 use BadMethodCallException;
 use Lcobucci\Jose\Parsing;
-use Lcobucci\JWT\Claim\Factory as ClaimFactory;
 use Lcobucci\JWT\Signer\Key;
 
 /**
- * This class makes easier the token creation process
- *
  * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
- * @since 0.1.0
+ * @since 4.0.0
  */
-class Builder
+interface Builder
 {
     /**
-     * The token header
-     *
-     * @var array
-     */
-    private $headers;
-
-    /**
-     * The token claim set
-     *
-     * @var array
-     */
-    private $claims;
-
-    /**
-     * The token signature
-     *
-     * @var Signature
-     */
-    private $signature;
-
-    /**
-     * The data encoder
-     *
-     * @var Parsing\Encoder
-     */
-    private $encoder;
-
-    /**
-     * The factory of claims
-     *
-     * @var ClaimFactory
-     */
-    private $claimFactory;
-
-    /**
-     * Initializes a new builder
-     *
-     * @param Parsing\Encoder $encoder
-     * @param ClaimFactory $claimFactory
+     * Appends a new audience
      */
-    public function __construct(
-        Parsing\Encoder $encoder,
-        ClaimFactory $claimFactory
-    ) {
-        $this->encoder = $encoder;
-        $this->claimFactory = $claimFactory;
-        $this->headers = ['typ'=> 'JWT', 'alg' => 'none'];
-        $this->claims = [];
-    }
-
-    /**
-     * Configures the audience
-     *
-     * @param string|array $audience
-     * @param bool $addHeader
-     *
-     * @return Builder
-     */
-    public function canOnlyBeUsedBy(string $audience, bool $addHeader = false): Builder
-    {
-        $audiences = isset($this->claims['aud'])
-            ? array_merge($this->claims['aud']->getValue(), [$audience])
-            : [$audience];
-
-        return $this->setRegisteredClaim(
-            'aud',
-            array_values(array_map('strval', $audiences)),
-            $addHeader
-        );
-    }
+    public function canOnlyBeUsedBy(string $audience, bool $addHeader = false): Builder;
```

Comment:
Those names were defined on #113 we're just extracting an interface. Let's define on #150 and change them on a different PR.